pkg/providerconfig/types_test.go

```diff
@@ -28,32 +30,148 @@ func TestConfigVarStringUnmarshalling(t *testing.T) {
 
 func TestConfigVarBoolUnmarshalling(t *testing.T) {
 	jsonBool := []byte("true")
-	jsonString := []byte(`"true"`)
 	jsonMapBool := []byte(`{"value":true}`)
-	jsonMapString := []byte(`{"value":"true"}`)
 
 	expectedResult := ConfigVarBool{Value: true}
 
 	var jsonBoolTarget ConfigVarBool
-	var jsonStringTarget ConfigVarBool
 	var jsonMapBoolTarget ConfigVarBool
-	var jsonMapStringTarget ConfigVarBool
 
 	err := json.Unmarshal(jsonBool, &jsonBoolTarget)
 	if err != nil || !reflect.DeepEqual(expectedResult, jsonBoolTarget) {
 		t.Fatalf("Decoding raw bool into configVarBool failed! Error: '%v'", err)
 	}
-	err = json.Unmarshal(jsonString, &jsonStringTarget)
-	if err != nil || !reflect.DeepEqual(expectedResult, jsonStringTarget) {
-		t.Fatalf("Decoding raw string bool into configVarBool failed! Error: '%v'", err)
-	}
 	err = json.Unmarshal(jsonMapBool, &jsonMapBoolTarget)
 	if err != nil || !reflect.DeepEqual(expectedResult, jsonMapBoolTarget) {
 		t.Fatalf("Decoding map bool into configVarBool failed! Error: '%v'", err)
 	}
-	err = json.Unmarshal(jsonMapString, &jsonMapStringTarget)
-	if err != nil || !reflect.DeepEqual(expectedResult, jsonMapStringTarget) {
-		t.Fatalf("Decoding map string bool into configVarBool failed! Error: '%v'", err)
+}
+
+func TestConfigVarStringMarshalling(t *testing.T) {
+	t.Parallel()
```

Comment:
this is a little bit unusual, because those test take fraction of second to complete :)

Reply:
Well its probably not strictly required but is there any drawback in using it? If I understand the docs correctly, golang will run everything sequentually that does not have that call above, since this test can be parallelized it doesn't harm or am I wrong there?

Reply:
I think the time needed to create and schedule a separate `goroutine` is longer than just let it run :)

All tests in the same package are run sequentially.
Tests across packages are run concurrently.